lib/abci/handlers/beginBlockHandlerFactory.js

```diff
@@ -46,6 +50,24 @@ function beginBlockHandlerFactory(
       );
     }
 
+    const previousTransactions = blockExecutionDBTransactions.getPreviousTransactions();
+
+    await Promise.all(
```

Comment:
What if this block won't be committed for some reason. I guess we need to commit transactions only on blockCommit